internal/registration/registration.go

```diff
@@ -87,3 +96,26 @@ func (r *Registration) registerDeviceOnce() error {
 	}
 	return nil
 }
+
+func (r *Registration) Deregister(configuration models.DeviceConfigurationMessage) error {
```

Comment:
This method doesn't need a parameter (`configuration models.DeviceConfigurationMessage`).